src/databasetasks.cpp

```diff
@@ -71,11 +71,11 @@ void DatabaseTasks::runTask(const DatabaseTask& task)
 	bool success;
 	DBResult_ptr result;
 	if (task.store) {
-		result = db.storeQuery(task.query);
+		result = Database::storeQuery(task.query);
```

Comment:
The async database tasks used a different db connection (yes, we had two instances of  a class that is supposed to be a singleton...) than the dispatcher, so you've kinda killed their usefulness with this patch :/,